src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -489,6 +492,33 @@ private void doExecute() throws MojoExecutionException {
             }
         }
     }
+    
+    private void fillSensibleHeaderTypeDefaults() throws MojoExecutionException {
+        if (headerType == null) {
+            if (classPath != null && classPath.mainClass != null) {
+                if (classPath.mainClass.toLowerCase().contains("gui")) {
+                    getLog().warn("headerType not set, defaulting to \"gui\" because the main class is named " + classPath.mainClass);
+                    headerType = "gui";
+                    return;
+                }
+            }
+
+            getLog().warn("headerType not set, defaulting to \"console\"");
+            headerType = "console";
```

Comment:
Wouldn't be better to make the `headerType` attribute required with a default value like this?
```java
    @Parameter(required = true, defaultValue = "console")
    private String headerType;
```

Reply:
Sure, if you want a single default value with no 'magic' that's better